### DISCUSSION TITLE
Possible Error with AutoForm in Blazor?

### DISCUSSION THREAD

**Discussion by Ryan-Britton** [ServiceStack/Discuss#118] [Q&A]
Hi there,

I found what I think might be an error processing enums with the AutoForm control in Blazor.

I have the following enum :

```
public enum ConnectionType
{
    [Description("Unknown")]
    [EnumMember(Value="Unknown")]
    Unknown,

    [Description("Windows File System")]
    [EnumMember(Value="Windows File System")]
    WindowsFileSystem,

    [Description("Microsoft SharePoint Online")]
    [EnumMember(Value="Microsoft SharePoint Online")]
    MicrosoftSharePointOnline
}
```

and the following request : 

```
[DataContract]
public class CreateConnection
    : IReturn<IdResponse>, IPost, ICreateDb<Connection>
{
    [DataMember(Order=2)]
    [Required]
    public string Name { get; set; }

    [DataMember(Order=3)]
    public ConnectionType ConnectionType { get; set; }
}
```

when saving the Autform submission, I can see the following on line 154 of AutoFormBase.cs : 

![image](https://github.com/ServiceStack/Discuss/assets/124330161/f034e53c-c9f0-441b-9e32-b298e95b9cc8)

the returned value of "api" after the call to ApiFormSync processes, however - is the following: 

![image](https://github.com/ServiceStack/Discuss/assets/124330161/e5b11508-69a2-4e69-9d21-3cb5bab99002)

it seems like the return value of the enum might be getting lost in translation? The correct value is saved to the database, and the api services are standard AutoQuery services (no custom implementation)

**Comment by mythz** [comment]
If you have an active commercial license please report any issues to our [Issue Tracker](https://github.com/ServiceStack/Issues). If you don't we accept PR's for failing tests or a link to a stand-alone repro. It's more likely JSON serialization issue instead of an issue with a Blazor Component in which case only a repro for the AutoQuery API would be needed.
